src/Compilers/Core/Portable/Diagnostic/DiagnosticSeverity.cs

```diff
@@ -27,7 +27,12 @@ public enum DiagnosticSeverity
         /// <summary>
         /// Something not allowed by the rules of the language or other authority.
         /// </summary>
-        Error = 3
+        Error = 3,
+
+        /// <summary>
+        /// Something that can be potentially improved.
+        /// </summary>
+        Suggestion = 4,
```

Comment:
not sure how Error+Unecessary make sense if we consider unnecessary as characteristic rather than visual effect.

error can't be unnecessary. if we consider unnecessary as cue for visual effect than

hidden + suggestion also make sense. same as error, warning, info + suggestion.